pkg/liquidity-mining/test/VotingEscrowRemapper.test.ts

```diff
@@ -0,0 +1,560 @@
+            });
+
+            it('bridges the remapped address and clears the pre-existing one', async () => {
+              await remapper
+                .connect(local)
+                .setNetworkRemapping(local.address, other.address, chainId, { value: nativeFee });
+
+              const receipt = await (await doRemap(nativeFee * 2)).wait();
+              expectEvent.inIndirectReceipt(
+                receipt,
+                omniVotingEscrow.interface,
+                'SendUserBalance',
+                {
+                  user: local.address,
+                  chainId,
+                  refundAddress: caller.address,
+                },
+                omniVotingEscrow.address
+              );
+
+              expectEvent.inIndirectReceipt(
+                receipt,
+                omniVotingEscrow.interface,
+                'SendUserBalance',
+                {
+                  user: other.address,
+                  chainId,
+                  refundAddress: caller.address,
+                },
+                omniVotingEscrow.address
+              );
+            });
+          });
+        });
+      }
+
+      function itRevertsIfRemoteUserIsZero() {
+        it('reverts if remote address is zero', async () => {
+          await expect(
+            remapper.connect(caller).setNetworkRemapping(local.address, ZERO_ADDRESS, chainId)
+          ).to.be.revertedWith('Zero address cannot be used as remote user');
+        });
+      }
+    });
+  });
+
+  describe('clearNetworkRemapping', () => {
+    const chainId = 7;
+    const otherChainId = 43;
+    let caller: SignerWithAddress;
+    const doClearMap = async (value = 0) =>
+      remapper.connect(caller).clearNetworkRemapping(local.address, chainId, { value });
+
+    function itClearsNetworkRemapping() {
+      it('clears existing local to remote mapping in target chain ID', async () => {
+        await doClearMap();
+        expect(await remapper.getRemoteUser(local.address, chainId)).to.be.eq(ZERO_ADDRESS);
+      });
+
+      it('does not clear existing local to remote mapping in other chain ID', async () => {
+        await doClearMap();
+        expect(await remapper.getRemoteUser(local.address, otherChainId)).to.be.eq(remote);
+      });
+
+      it('clears existing remote to local mapping in target chain ID', async () => {
+        await doClearMap();
+        expect(await remapper.getLocalUser(remote, chainId)).to.be.eq(ZERO_ADDRESS);
+      });
+
+      it('does not clear existing remote to local mapping in other chain ID', async () => {
+        await doClearMap();
+        expect(await remapper.getLocalUser(remote, otherChainId)).to.be.eq(local.address);
+      });
+
+      it('emits an AddressMappingUpdated event', async () => {
+        const tx = await doClearMap();
+
+        expectEvent.inReceipt(await tx.wait(), 'AddressMappingUpdated', {
+          localUser: local.address,
+          remoteUser: ZERO_ADDRESS,
+          chainId,
+        });
```

Comment:
I wonder if we should also emit an event indicating that `remote` no longer has a local

Reply:
Hmm if you overwrite an existing remapping, there's a remote that no longer maps to a local too.
We could do it, but maybe for clarity we should use two different events. WDTY?

Reply:
Added in https://github.com/balancer/balancer-v2-monorepo/pull/2174/commits/36326229a89ca1248d1838df02c493be0cda42bc.